### PR TITLE
Rockchip updates / fixes

### PR DESCRIPTION
**Pull request by knaerzche**
Some fixes / additions for Rockchip platform:

* fixed YUV output for RK3288 / RK3399
* fixed HEVC decoding
* added HW reset for video decoder (rkvdec) which now avoids (the rare) lockups more reliable
* aligend HDMI sound-card names in device trees with alsa config, to allow HDMI audio passthrough and also added a patch from @HiassofT which fixes ELD constraints - passthough seems to be working now in allmost all cases (I have) - also AC3
* added ES8316 codec (used on many RK3399 boards) to soundconfig which unmutes headphone jack per default
* enabled NVMe support for RK3399
* some minor linux-config alignments / additions

### CODE REVIEW

File: projects/Rockchip/patches/linux/default/linux-1003-for-libreelec.patch [hunk 1]
```diff
@@ -726,3 +764,91 @@ index 70ab4fbdc23e..4dbc370e538f 100644
  		cec_transmit_attempt_done(adap, cec->tx_status);
  	}
  	if (cec->rx_done) {
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Matthias Reichl <hias@horus.com>
+Date: Fri, 19 Mar 2021 12:14:17 +0100
+Subject: [PATCH] ALSA: pcm: fix ELD constraints for some compressed audio
+ formats
+
```
Comment (jernejsk) [inline projects/Rockchip/patches/linux/default/linux-1003-for-libreelec.patch:772]: This should imo be moved to common folder for mainline Linux. It's useful for all platforms...

Comment (knaerzche) [reply projects/Rockchip/patches/linux/default/linux-1003-for-libreelec.patch:772]: I also thought so, but I'm not sure if it messes something up for generic. would it @HiassofT?

Comment (HiassofT) [reply projects/Rockchip/patches/linux/default/linux-1003-for-libreelec.patch:772]: No objections moving this to default kernel patches.

Generic uses the Intel HDA ALSA driver, not ASoC, so won't be affected by the change.

**Comment by knaerzche** [comment]
moved ELD constraints fix patch common linux patches

**Review by chewitt — APPROVED** [review]

**Pull request merged**